### DISCUSSION TITLE
Using bpftrace helpers

### DISCUSSION THREAD

**Discussion by techneophyte** [bpftrace/bpftrace#3272] [General]
Hello everyone, for getting the current process `pid`, I am trying to use `bpf_get_current_pid_tgid()` BPF helper inside a bt script. I am seeing an error
`ERROR: Unknown function: bpf_get_current_pid_tgid`

Can someone please help me on using these bpf helpers inside a bpf trace script or another way of getting current process id?

**Comment by ajor** [comment]
bpftrace doesn't expose helpers directly. It provides various [functions](https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#functions) and [variables](https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#builtins) to give access to their functionality instead.

You can get the PID of the current process by using the builtin `pid` variable in bpftrace:
```
BEGIN
{
  printf("%d\n", pid);
}
```
